Edit review endpoint

tj28601 opened the pull request:
added update route to reviews w accompanying tests

JauntyJames committed:
Added edit method to reviews controller

Commit:
added update route for reviews

dpickett commented on app/controllers/api/v1/reviews_controller.rb:
```diff
@@ -5,18 +5,35 @@ def create
       # Design descision to be made: will the react element need the wizard?
       if new_review.save
-        render json: { wizard: wizard , review: review_params }
+        render json: { wizard: wizard, review: review_params }
```
RESTful convention wise, we wouldn't necessarily return the wizard in the response body when we _create_ a review.

dpickett commented on app/controllers/api/v1/reviews_controller.rb:
```diff
@@ -5,18 +5,35 @@ def create
+        render json: { wizard: wizard, review: review_params }
       else
-        render json: { wizard: wizard , review: review_params, errors: new_review.errors.full_messages }, status: :unprocessable_entity
+        render json: { wizard: wizard, review: review_params, errors: new_review.errors.full_messages }, status: :unprocessable_entity
```
we would probably just respond with the errors here. Good job on the status code.

dpickett commented on app/controllers/api/v1/reviews_controller.rb:
```diff
@@ -5,18 +5,35 @@ def create
       end
     else
-      render json: { wizard: wizard , review: review_params, errors: "Access Denied" }, status: :unprocessable_entity
+      render json: { wizard: wizard, review: review_params, errors: "Access Denied" }, status: :unprocessable_entity
```
what is an alternative status code you can use here to tell the user that authentication is required to RESTfully represent that response?

dpickett commented on app/controllers/api/v1/reviews_controller.rb:
```diff
@@ -5,18 +5,35 @@ def create
+  def update
+
+    review = Review.find(review_params[:id])
+    if current_user.id == review_params[:user_id].to_i
```
`review_params[:user_id]` is easily hackable - you should use the user_id that is fetched from the database to authorize the user here.

dpickett commented on app/controllers/api/v1/reviews_controller.rb:
```diff
@@ -5,18 +5,35 @@ def create
 
   def review_params
-    params.require(:review).permit(:body, :wizard_id, :user_id, :rating)
+    params.require(:review).permit(:body, :wizard_id, :user_id, :rating, :id)
```
you should not allow the user to update the `id`. I would remove that from the allow params here

iBrooks committed:
Removed returning of wizard in the create response
removed returning of wizard and reviews if user is no signed in and
tries to create
updated error code to :forbidden
switched to using user_id found in DB
removed user_id from sterilized params

iBrooks committed:
Merge branch 'edit-review-endpoint' of https://github.com/prophetsfall/wizards_reviews into edit-review-endpoint

iBrooks committed:
resolved testing errors
